nightingale_navigation/hallway_planner/src/geometry_primitives.cpp

```diff
@@ -0,0 +1,124 @@
+/*********************************************************************
+ * Software License Agreement (MIT License)
+ * Author: Younes Reda
+ *********************************************************************/
+
+#include <hallway_planner/geometry_primitives.hpp>
+#include <cmath>
+
+namespace hallway_planner::geometry_primitives
+{
+
+Point::Point(const uint32_t &X, const uint32_t &Y) : x_(X), y_(Y) {}
+
+bool Point::operator==(const Point &p) const
+{
+    return (x_==p.x()) && (y_==p.y());
+}
+
+bool Point::operator!=(const Point &p) const
+{
+    return !(operator==(p));
+}
+
+int64_t Point::x() const
+{
+    return static_cast<int64_t>(x_);
+}
+
+int64_t Point::y() const
+{
+    return static_cast<int64_t>(y_);
+}
+
+Point::operator cv::Point() const
+{
+    return cv::Point(x_,y_);
+}
+
+double euclidean_distance(const Point &a, const Point &b)                                            
+{
+    const auto dy = a.y()-b.y();
+    const auto dx = a.x()-b.x();
+    return std::sqrt(dy*dy+dx*dx);                                                                   
+}
+
+Line::Line(const Point &A, const Point &B) : a_(A), b_(B),
+                                             slope_(Line::calculate_slope(a_,b_)),
+                                             y_intercept_(Line::calculate_y_intercept(slope_,a_)),
+                                             theta_(Line::calculate_theta(a_,b_)),
+                                             rho_(Line::calculate_rho(a_,theta_))
+{}
+
+double Line::length() const
+{
+    return euclidean_distance(a_,b_);
+}
+
+double Line::calculate_slope(const Point &a, const Point &b)
+{
+    // let slope be positive since choosing which point is first is arbitrary
+    const double num = a.y()-b.y();
```

Comment:
just curious and not feedback but why use double over auto here?

Reply:
the y() function returns an int. if i use auto, num will be an int. when i do division, it will be integer division which would be wrong